test/inputs-dovetail-bytes-test.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const support = require('./support');
+const bigquery = require('../lib/bigquery');
+const DovetailBytes = require('../lib/inputs/dovetail-bytes');
+
+describe('dovetail-bytes', () => {
+
+  let bytes = new DovetailBytes();
+
+  it('recognizes bytes records', () => {
+    expect(bytes.check({})).to.be.false;
+    expect(bytes.check({type: 'anything'})).to.be.false;
+    expect(bytes.check({type: 'bytes'})).to.be.true;
+    expect(bytes.check({type: 'segmentbytes'})).to.be.true;
+  });
+
+  it('formats download byte inserts', () => {
+    const record = bytes.formatDownload({
+      timestamp: 1490827132999,
+      request_uuid: 'the-uuid',
+      bytes_downloaded: 123,
+      seconds_downloaded: 1.23,
+      percent_downloaded: 0.12,
+    })
+    expect(record).to.have.keys('insertId', 'json');
+    expect(record.insertId).to.equal('bytes-the-uuid');
+    expect(record.json).to.have.keys('timestamp', 'request_uuid', 'bytes', 'seconds', 'percent');
+    expect(record.json.timestamp).to.equal(1490827132);
+    expect(record.json.request_uuid).to.equal('the-uuid');
+    expect(record.json.bytes).to.equal(123);
+    expect(record.json.seconds).to.equal(1.23);
+    expect(record.json.percent).to.equal(0.12);
+  });
+
+  it('formats impression byte inserts', () => {
+    const record = bytes.formatImpression({
+      timestamp: 1490827132999,
+      request_uuid: 'the-uuid',
+      segment_index: 2,
+      bytes_downloaded: 123,
+      seconds_downloaded: 1.23,
+      percent_downloaded: 0.12,
+    })
+    expect(record).to.have.keys('insertId', 'json');
+    expect(record.insertId).to.equal('bytes-the-uuid-2');
```

Comment:
Are these ids reused anywhere?

Reply:
Just in this function.  If you use the same `insertId` within a few minutes, BigQuery will ignore the 2nd insert.  So if we fail on a batch of kinesis records, it's okay to re-execute them (within a few minutes) ... because even if we end up inserting dups, they should get ignored.